modules/storage/src/service/fs.rs

```diff
@@ -0,0 +1,225 @@
+use crate::service::{StorageBackend, StoreError};
+use anyhow::Context;
+use bytes::Bytes;
+use futures::{Stream, StreamExt};
+use sha2::{digest::Output, Digest, Sha256};
+use std::{
+    fmt::Debug,
+    io::ErrorKind,
+    io::SeekFrom,
+    path::{Path, PathBuf},
+    pin::pin,
+};
+use tempfile::{tempdir, tempfile, TempDir};
+use tokio::{
+    fs::{create_dir, create_dir_all, File},
+    io::{AsyncSeekExt, AsyncWriteExt},
+};
+use tokio_util::io::ReaderStream;
+
+/// A filesystem backed store
+///
+/// ## Layout
+///
+/// The layout of the filesystem is as follows:
+///
+/// ```ignore
+/// <base>/
+///   content/
+///     <digest[0..2]>/
+///       <digest[2..4]>/
+///         <digest> # file
+/// ```
+///
+/// The idea behind that is to limit the number of directory entries. For some filesystems,
+/// the performance can degrade is directories get too big (have too many entries).
+///
+/// This layout limits the number of entries on the first two layers to 256, and limits the chance
+/// of a file ending up in the same directory by 65536. Assuming an average distribution of hashes,
+/// that would allow us to store 16777216 files, until we hit the cap of 256 entries in the lowest
+/// lever directories. Should we need to increase that limit, we could easily add an additional
+/// layer.
+#[derive(Clone, Debug)]
+pub struct FileSystemBackend {
+    content: PathBuf,
+}
+
+const NUM_LEVELS: usize = 2;
+
+impl FileSystemBackend {
+    pub async fn new(base: impl Into<PathBuf>) -> anyhow::Result<Self> {
```

Comment:
See my pending PR for `trustd`. We should maybe default to `.trustify/...whatever` for the Simple Happy PM Use-Case?

Reply:
(not this PR though)

Reply:
Should this go in #121 ?